### DISCUSSION TITLE
Forge installation error

### DISCUSSION THREAD

**Discussion by Wilfychep** [Cyfrin/foundry-full-course-cu#666] [Q&A]
Hi, I get this error during fundryup installation. Forge path. 

/... foundryup: installing foundry (version nightly, tag nightly-06a17bfe3265209a854353337cc4277c1a4c73d1)
foundryup: downloading latest forge, cast, anvil, and chisel
######################################################################################### 100.0%
foundryup: downloading manpages
######################################################################################### 100.0%
foundryup: installed - forge 0.2.0 (06a17bf 2023-08-19T00:14:54.453240995Z)
foundryup: warning: 
There are multiple binaries with the name 'forge' present in your 'PATH'.
This may be the result of installing 'forge' using another method,
like Cargo or other package managers.
You may need to run 'rm /usr/bin/forge' or move '/home/chelel/.foundry/bin'
in your 'PATH' to allow the newly installed version to take precedence!

foundryup: installed - cast 0.2.0 (06a17bf 2023-08-19T00:14:54.444477665Z)
foundryup: installed - anvil 0.2.0 (06a17bf 2023-08-19T00:14:54.448798525Z)
foundryup: installed - chisel 0.2.0 (06a17bf 2023-08-19T00:14:54.457250529Z)
foundryup: done!     .../

**Comment by theirrationalone** [comment]
Heyyy @Wilfychep !, You may wanna follow these steps [discussion](https://github.com/Cyfrin/foundry-full-course-f23/discussions/644#discussioncomment-6730003)...

After this migitgation: If you Still getting issues! Feel Free to ask.....😉

**Reply by Wilfychep** [reply]
Hi. Tried the steps but still not successful.

**Reply by theirrationalone** [reply]
Show me the error logs and new binaries screenshot...

**Reply by Wilfychep** [reply]
As follows.../   My project is known as Koffie

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx
 
 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development 
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Repo       : https://github.com/foundry-rs/
Book       : https://book.getfoundry.sh/                      
Chat       : https://t.me/foundry_rs/                         
Support    : https://t.me/foundry_support/
Contribute : https://github.com/orgs/foundry-rs/projects/2/

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

foundryup: installing foundry (version nightly, tag nightly-06a17bfe3265209a854353337cc4277c1a4c73d1)
foundryup: downloading latest forge, cast, anvil, and chisel
################################################################################################ 100.0%
foundryup: downloading manpages
################################################################################################ 100.0%
foundryup: installed - forge 0.2.0 (06a17bf 2023-08-19T00:14:54.453240995Z)
foundryup: warning: 
There are multiple binaries with the name 'forge' present in your 'PATH'.
This may be the result of installing 'forge' using another method,
like Cargo or other package managers.
You may need to run 'rm /usr/bin/forge' or move '/home/chelel/.foundry/bin'
in your 'PATH' to allow the newly installed version to take precedence!

foundryup: installed - cast 0.2.0 (06a17bf 2023-08-19T00:14:54.444477665Z)
foundryup: installed - anvil 0.2.0 (06a17bf 2023-08-19T00:14:54.448798525Z)
foundryup: installed - chisel 0.2.0 (06a17bf 2023-08-19T00:14:54.457250529Z)
foundryup: done!
chelel@Chepkwony:~/koffie$

**Reply by theirrationalone** [reply]
Go back to `.bin` folder 📂 and delete all files whose name have forge included. Then again try to run foundryup

Your bin path is this: `/home/chelel/.foundry/bin`

**Reply by Wilfychep** [reply]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/115069076/4e568880-908c-4f76-aaf3-8bbf8e05ba98)

**Reply by Wilfychep** [reply]
Above is to show the path.   I delete those folders and run foundryup i get the same results

**Reply by theirrationalone** [reply]
![IMG_20230820_000256.jpg](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/1717885d-d6fd-4f2f-8e5e-a10941a0fec8)

Delete those two and go one step back and then enter into `.cache` folder and delete everything in that but don't delete `.cache` itself.. Next try again with foundryup

**Reply by Wilfychep** [reply]
Hi.. Still not successful.  
Below is my path ,maybe you can  help point out if there is something I am missing.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/115069076/433699a3-5476-4543-9b88-8cc9024763cd)

**Reply by Wilfychep** [reply]
Cleared things in the cache as well

**Reply by Wilfychep** [reply]
Under the cache ,there is this yarn folder that cant seem to delete completely.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/115069076/16449713-0bb0-4819-b54c-93e23d500773)

**Reply by theirrationalone** [reply]
I'm Giving my Structure and Directions too into screen shots, Please follow them....

![dgit disss](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/3f2024f1-7c8e-4de1-8e41-8395737af4cc)

<br>
<br>

![dgit diss2](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/cec1c5fa-4aaa-4344-a870-d5fdff1642bf)

<br>
<br>

![dgit disss3](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/b1c3c5c4-cd64-423b-91cd-20d2156772e8)

**Reply by Wilfychep** [reply]
Hey.. I understood the steps even without the screenshots and followed to the latter and I am grateful for your assistance. I think my problem lies with the .cache since there's a specific folder that i cant seem to get rid of a yarn folder that says it cant be found, it has some three folders inside(npm)

**Reply by theirrationalone** [reply]
Your Logs telling that you have two binaries named `forge`, So Please try to find two identical forge binaries and delete one of them
Or

I would like to suggest you the best mitigation to tackle with this issue: Please,
1. delete/uninstall `wsl` from your `PC`.
2. Update VS-Code.
3.  Make/Install your foundry environment again from very beginning.
